### PR TITLE
Capture referrer during a shop purchase

### PR DESCRIPTION
**Pull request by minkyngkm**
Open this PR again as this [PR](https://github.com/canonical/ubuntu.com/pull/12520) has been closed when the checkout branch was deleted.

## Done

-Capture referrer during a shop purchase

## QA

- Run locally 
- print `flask.session` somewhere e.g. line 77 in `/shop/decorator.py`
- Go to /pro/subscribe?referrer=canonical
- Check `'referrer': 'canonical'` is saved in session

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-1790

### CODE REVIEW

**Review by jpmartinspt — APPROVED** [review]
Lgtm!

File: static/js/src/advantage/subscribe/react/components/PaymentButton/PaymentButton.tsx [hunk 1]
```diff
@@ -2,9 +2,9 @@ import React, { useContext } from "react";
 import { Button } from "@canonical/react-components";
 import { FormContext } from "../../utils/FormContext";
 
-export default function PaymentButton() {
+export default function PaymentButton(referrer: { referrer: string | null }) {
   const { quantity, product } = useContext(FormContext);
-
+  const referrerValue = referrer ? referrer.referrer : "";
```
Comment (anthonydillon) [inline static/js/src/advantage/subscribe/react/components/PaymentButton/PaymentButton.tsx:7]: Is empty string for none here risky?

**Comment by anthonydillon** [comment]
@minkyngkm the PR description here could be more detailed. In particular, the QA steps.

File: static/js/src/advantage/subscribe/react/components/PaymentButton/PaymentButton.tsx (hunk 1 above)
Comment (minkyngkm) [reply static/js/src/advantage/subscribe/react/components/PaymentButton/PaymentButton.tsx]: This can be removed! Thanks 👍

**Review by jpmartinspt — APPROVED** [review]

**Pull request merged**